api/nodes/views.py

```diff
@@ -142,12 +142,63 @@ def get_default_queryset(self):
     def get_queryset(self):
         return self.get_queryset_from_request()
 
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class NodeContributorDetail(generics.RetrieveUpdateDestroyAPIView, NodeMixin):
+    """Detail of a contributor for a node.
+
+    Allows for a user to view, remove a contributor from, and change bibliographic and admin status for a node.
+    """
+    serializer_class = NodeContributorDetailSerializer
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        Contributor,
+    )
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        node = self.get_node()
+        user_lookup_url_kwarg = 'user_id'
+        user = get_object_or_404(User, self.kwargs[user_lookup_url_kwarg])
+        # May raise a permission denied
+        self.check_object_permissions(self.request, user)
+        if user not in node.contributors:
+            raise NotFound('{} cannot be found in the list of contributors.'.format(user))
+        user.permissions = node.get_permissions(user)
+        user.bibliographic = node.get_visible(user)
+        user.node_id = node._id
+        return user
+
+    # overrides DestroyAPIView
+    def perform_destroy(self, instance):
+        node = self.get_node()
+        current_user = self.request.user
+        auth = Auth(current_user)
+        if not self.has_multiple_admins(node) and node.has_permission(instance, 'admin'):
+            raise ValidationError("Must have at least one registered admin contributor")
+        if len(node.visible_contributors) == 1 and node.get_visible(instance):
+            raise ValidationError("Must have at least one visible contributor")
+        node.remove_contributor(instance, auth)
+
+    def has_multiple_admins(self, node):
```

Comment:
You shouldn't need this in both the view and the serializer. In a pinch, you could define in one place and import, but I expect you could just pass the results from the view to the serializer, yes?